### PR TITLE
ADEN-10505 Introduce Startup 2.0 to all platforms

### PR DESCRIPTION
**Pull request by Bielik20**

### CODE REVIEW

File: platforms/futhead/futhead-platform.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+import { FutheadTemplatesSetup } from './templates/futhead-templates.setup';
+
+export class FutheadPlatform {
+	constructor(private pipeline: ProcessPipeline, private noAdsDetector: NoAdsDetector) {}
+
+	execute(): void {
+		// Config
+		this.pipeline.add(
+			() => context.extend(basicContext),
+			parallel(InstantConfigSetup, () => ensureGeoCookie().then(() => bootstrapAndGetConsent())),
```
Comment (fraszczakszymon) [inline platforms/futhead/futhead-platform.ts:40]: I believe ensureGeoCookie should happen before `InstantConfigSetup`, am I wrong?

Comment (fraszczakszymon) [reply platforms/futhead/futhead-platform.ts:40]:
```suggestion
			() => ensureGeoCookie(),
			parallel(InstantConfigSetup, () => bootstrapAndGetConsent()),
```

File: platforms/gamepedia/gamepedia-platform.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+import { GamepediaTemplatesSetup } from './setup/templates/gamepedia-templates.setup';
+
+export class GamepediaPlatform {
+	constructor(private pipeline: ProcessPipeline, private noAdsDetector: NoAdsDetector) {}
+
+	execute(): void {
+		// Config
+		this.pipeline.add(
+			() => context.extend(basicContext),
+			parallel(InstantConfigSetup, () => ensureGeoCookie().then(() => bootstrapAndGetConsent())),
```
Comment (fraszczakszymon) [inline platforms/gamepedia/gamepedia-platform.ts:40]: Ditto
```suggestion
			() => ensureGeoCookie(),
			parallel(InstantConfigSetup, () => bootstrapAndGetConsent()),
```

File: platforms/muthead/muthead-platform.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+import { MutheadTemplatesSetup } from './templates/muthead-templates.setup';
+
+export class MutheadPlatform {
+	constructor(private pipeline: ProcessPipeline, private noAdsDetector: NoAdsDetector) {}
+
+	execute(): void {
+		// Config
+		this.pipeline.add(
+			() => context.extend(basicContext),
+			parallel(InstantConfigSetup, () => ensureGeoCookie().then(() => bootstrapAndGetConsent())),
```
Comment (fraszczakszymon) [inline platforms/muthead/muthead-platform.ts:40]: Ditto
```suggestion
			() => ensureGeoCookie(),
			parallel(InstantConfigSetup, () => bootstrapAndGetConsent()),
```

File: platforms/ucp-hydra/ucp-hydra-platform.ts [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+			() => context.extend(basicContext),
+			parallel(InstantConfigSetup, () => bootstrapAndGetConsent()),
+			UcpHydraIocSetup,
+			UcpWikiContextSetup,
+			() => context.set('state.isMobile', false),
+			UcpBaseContextSetup,
+			HydraSlotsContextSetup,
+			UcpTargetingSetup,
+			UcpGamepediaPrebidConfigSetup,
+			GamepediaA9ConfigSetup,
```
Comment (fraszczakszymon) [inline platforms/ucp-hydra/ucp-hydra-platform.ts:47]: Why do we name these two in a different way?

File: platforms/ucp-minerva/ucp-minerva-platform.ts [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+			() => context.extend(basicContext),
+			parallel(InstantConfigSetup, () => bootstrapAndGetConsent()),
+			UcpMinervaIocSetup,
+			UcpWikiContextSetup,
+			() => context.set('state.isMobile', true),
+			UcpBaseContextSetup,
+			MinervaSlotsContextSetup,
+			UcpTargetingSetup,
+			UcpGamepediaPrebidConfigSetup,
+			GamepediaA9ConfigSetup,
```
Comment (fraszczakszymon) [inline platforms/ucp-minerva/ucp-minerva-platform.ts:47]: Ditto

**Review by fraszczakszymon — CHANGES_REQUESTED** [review]
Well done 💎 . I can see one issue only: we need to fetch geo data on FUT/MUT/GP before ICBM is configured.

File: platforms/futhead/futhead-platform.ts (hunk 1 above)
Comment (Bielik20) [reply platforms/futhead/futhead-platform.ts:40]: Great catch

File: platforms/ucp-hydra/ucp-hydra-platform.ts (hunk 4 above)
Comment (Bielik20) [reply platforms/ucp-hydra/ucp-hydra-platform.ts:47]: I don't know... my guess is someone changed one and forgot about the other 🤷 
I changed it though.

**Review by fraszczakszymon — APPROVED** [review]

**Pull request merged**